partners_api/booking_api.cpp

```diff
@@ -1,37 +1,243 @@
 #include "partners_api/booking_api.hpp"
 
+#include "platform/http_client.hpp"
+#include "platform/platform.hpp"
+
 #include "base/gmtime.hpp"
 #include "base/logging.hpp"
+#include "base/thread.hpp"
 
-#include "std/chrono.hpp"
+#include "std/initializer_list.hpp"
+#include "std/iomanip.hpp"
 #include "std/iostream.hpp"
 #include "std/sstream.hpp"
+#include "std/utility.hpp"
 
 #include "3party/jansson/myjansson.hpp"
 
 #include "private.h"
 
-char constexpr BookingApi::kDefaultCurrency[1];
+namespace
+{
+using namespace platform;
+using namespace booking;
+
+string const kBookingApiBaseUrl = "https://distribution-xml.booking.com/json/bookings";
+string const kExtendedHotelInfoBaseUrl = "http://hotels.milchakov.map6.devmail.ru/getDescription";
+string const kPhotoOriginalUrl = "http://aff.bstatic.com/images/hotel/max500/";
+string const kPhotoSmallUrl = "http://aff.bstatic.com/images/hotel/max300/";
 
-BookingApi::BookingApi() : m_affiliateId(BOOKING_AFFILIATE_ID), m_testingMode(false)
+bool RunSimpleHttpRequest(bool const needAuth, string const & url, string & result)
 {
-  stringstream ss;
-  ss << BOOKING_KEY << ":" << BOOKING_SECRET;
-  m_apiUrl = "https://" + ss.str() + "@distribution-xml.booking.com/json/bookings.";
+  HttpClient request(url);
+
+  if (needAuth)
+    request.SetUserAndPassword(BOOKING_KEY, BOOKING_SECRET);
+
+  if (request.RunHttpRequest() && !request.WasRedirected() && request.ErrorCode() == 200)
+  {
+    result = request.ServerResponse();
+    return true;
+  }
+  return false;
 }
 
-string BookingApi::GetBookHotelUrl(string const & baseUrl, string const & /* lang */) const
+string MakeApiUrl(string const & func, initializer_list<pair<string, string>> const & params,
+                  bool testing)
 {
-  return GetDescriptionUrl(baseUrl) + "#availability";
+  ostringstream os;
+  os << kBookingApiBaseUrl << "." << func << "?";
+
+  bool firstRun = true;
+  for (auto const & param : params)
+  {
+    if (firstRun)
+    {
+      firstRun = false;
+      os << "";
+    }
+    else
+    {
+      os << "&";
+    }
+    os << param.first << "=" << param.second;
+  }
+
+  if (testing)
```

Comment:
So, when there is no params at all, you don't need to add '&' before show_test, isn't it?